tests/test-validate-app-receipt.js

```diff
@@ -85,11 +89,13 @@ describe('fxpay.validateAppReceipt()', function() {
 
     var validator = new helper.ReceiptValidator();
 
-    fxpay.validateAppReceipt(function(error, productInfo) {
-      assert.instanceOf(error, fxpay.errors.InvalidReceipt);
-      assert.typeOf(productInfo, 'object');
+    fxpay.validateAppReceipt().then(function() {
+      done(Error('unexpected success'));
+    }).catch(function(reason) {
+      assert.instanceOf(reason.error, fxpay.errors.InvalidReceipt);
```

Comment:
If something here throws and we fall-through to the next catch. Will that cause a test failure?